lib/python/rose/config_editor/nav_panel_menu.py

```diff
@@ -65,7 +65,7 @@ def add_dialog(self, base_ns):
             else:
                 sections = self.data.helper.get_sections_from_namespace(
                     base_ns)
-                if sections == []:
+                if not len(sections):
```

Comment:
PEP8 says no:

> For sequences, (strings, lists, tuples), use the fact that empty sequences are false.
```python
Yes: if not seq:
     if seq:

No:  if len(seq):
     if not len(seq):
```

Reply:
See Sadie's previous comment... I believe there's a check elsewhere for `None`. At this step we are validating only list emptiness if I understand it correct.

And I tend to agree with what pep8/pycodestyle/black/etc say most time, but sometimes I have to review and then simply ignore pep8/et al.

e.g. of a warning that may change later (by the maintainer of TornadoI believe: https://github.com/PyCQA/pycodestyle/issues/815).

Reply:
Sorry I missed the bit about the check elsewhere for `None`.
Happy to approve if I know you know about the PEP8 question and have deliberately ignored it, although I'd like to understand why this is more desirable than PEP8's preference, which as far as I can see involves two fewer compiled instructions.

```python
>>> dis("a = []; not a")
  1           0 BUILD_LIST               0
              3 STORE_NAME               0 (a)
              6 LOAD_NAME                0 (a)
              9 UNARY_NOT
             10 POP_TOP
             11 LOAD_CONST               0 (None)
             14 RETURN_VALUE
>>> dis("a = []; not len(a)")
  1           0 BUILD_LIST               0
              3 STORE_NAME               0 (a)
              6 LOAD_NAME                1 (len)
              9 LOAD_NAME                0 (a)
             12 CALL_FUNCTION            1 (1 positional, 0 keyword pair)
             15 UNARY_NOT
             16 POP_TOP
             17 LOAD_CONST               0 (None)
             20 RETURN_VALUE

```